ssl/statem/statem_lib.c

```diff
@@ -1004,6 +1005,73 @@ int ssl_choose_server_version(SSL *s, CLIENTHELLO_MSG *hello)
         break;
     }
 
+    suppversions = tls_get_extension_by_type(hello->pre_proc_exts,
+                                             hello->num_extensions,
+                                             TLSEXT_TYPE_supported_versions);
+
```

Comment:
I'm not sure, I can't see anything in the spec about what to do in that case. It does say this:

"If the extension is not present, they MUST negotiate TLS 1.2 or prior as specified in [RFC5246], even if ClientHello.legacy_version is 0x0304 or later."

i.e. if hello->version == TLSv1.3, but supported_versions is not present then negotiate TLS1.2. 

I think we might encounter implementations which implement <=TLS1.2 but still send supported_versions....it doesn't say anywhere that you shouldn't do that.

In a conversation I had with @davidben he said "As for servers, the intent was that if you see supported_versions, you totally ignore ClientHello.version and evaluate that other list."